xtable-core/src/main/java/org/apache/xtable/iceberg/IcebergSchemaExtractor.java

```diff
@@ -146,10 +146,14 @@ private List<Types.NestedField> convertFields(
     List<Integer> ids =
         schema.getFields().stream()
             .map(
-                field ->
-                    field.getFieldId() == null
-                        ? fieldIdTracker.incrementAndGet()
-                        : field.getFieldId())
+                field -> {
+                  int id =
+                      field.getFieldId() == null
```

Comment:
If the fieldId is set, we need to use that ID. That is how we are able to handle renames in the schema. The reader will lookup the column based on the ID.